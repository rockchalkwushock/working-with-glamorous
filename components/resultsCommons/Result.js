import glamorous from 'glamorous'

const StyledResult = glamorous.li({
  color: 'green'
})

const StyledLink = glamorous.a({
  color: 'orange',
  ':visited': {
    color: 'orange'
  },
  ':hover': {
    color: 'green'
  }
})

StyledLink.displayName = 'StyledLink'
StyledResult.displayName = 'StyledResult'

const Result = ({ link, title }) =>
  <StyledResult>
    <StyledLink href={link}>
      {title}
    </StyledLink>
  </StyledResult>

export default Result
