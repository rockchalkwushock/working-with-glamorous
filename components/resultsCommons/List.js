import glamorous from 'glamorous'

import Result from './Result'

const StyledList = glamorous.ul({
  lineHeight: '1.5',
  listStyle: 'none',
  margin: '0',
  padding: '0',
  paddingBottom: '2em'
})

StyledList.displayName = 'StyledList'

const List = ({ questions }) =>
  <StyledList>
    {questions.map(q =>
      <Result key={Math.random()} link={q.link} title={q.title} />
    )}
  </StyledList>

export default List
