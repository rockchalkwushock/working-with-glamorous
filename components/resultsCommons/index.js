import List from './List'
import Result from './Result'

export { List, Result }
