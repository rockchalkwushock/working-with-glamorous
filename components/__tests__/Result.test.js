import { shallow, render, mount } from 'enzyme'

import { Result } from '../resultsCommons'

describe('Result Component', () => {
  let ui
  test('should render to the DOM', () => {
    ui = <Result />
    expect(shallow(ui)).toMatchSnapshotWithGlamor(`enzyme.shallow`)
    expect(mount(ui)).toMatchSnapshotWithGlamor(`enzyme.mount`)
    expect(render(ui)).toMatchSnapshotWithGlamor(`enzyme.render`)
  })
})
