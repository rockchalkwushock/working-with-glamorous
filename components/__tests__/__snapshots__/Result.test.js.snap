// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme.mount 1`] = `
.css-1xzrot3,
[data-css-1xzrot3] {
  color: green;
}

.css-v5siu3,
[data-css-v5siu3] {
  color: orange;
}

.css-v5siu3:visited,
[data-css-v5siu3]:visited {
  color: orange;
}

.css-v5siu3:hover,
[data-css-v5siu3]:hover {
  color: green;
}

<Result>
  <StyledResult>
    <li
      className="css-1xzrot3"
    >
      <StyledLink>
        <a
          className="css-v5siu3"
        />
      </StyledLink>
    </li>
  </StyledResult>
</Result>
`;

exports[`enzyme.render 1`] = `
.css-1xzrot3,
[data-css-1xzrot3] {
  color: green;
}

.css-v5siu3,
[data-css-v5siu3] {
  color: orange;
}

.css-v5siu3:visited,
[data-css-v5siu3]:visited {
  color: orange;
}

.css-v5siu3:hover,
[data-css-v5siu3]:hover {
  color: green;
}

<li
  class="css-1xzrot3"
>
  <a
    class="css-v5siu3"
  />
</li>
`;

exports[`enzyme.shallow 1`] = `
<StyledResult>
  <StyledLink />
</StyledResult>
`;
